src/Products/ZCatalog/Catalog.py

```diff
@@ -874,11 +863,11 @@ def _sort_nbest_reverse(self, actual_result_count, result, rs,
                             sort_index, sort_index_length, sort_spec,
                             second_indexes_key_map):
         # Limit / sort results using N-Best algorithm in reverse (N-Worst?)
-        index_key_map = sort_index.documentToKeyMap()
-        keys = []
-        n = 0
-        best = None
         if sort_index_length == 1:
+            index_key_map = sort_index.documentToKeyMap()
```

Comment:
Ideally, `sort_nbest` and `sort_nbest_reverse` would be symmetric. Your change above seems to have reduced the symmetry.

You have refactored the multi-index sorting into a single method. I would go a step further and merge `sort_nbest` and `sort_nbest_reverse` into a single function (in my view those methods are also named unintuitively; the `heapq` module contains similar functions named `nsmallest` and `nlargest`, respectively - which would be much better names). I would use `functools.partial` to instantiate the proper sorting function from it.